pkg/model/stack.go

```diff
@@ -12,33 +12,31 @@ type Stack struct {
 	Name      string    `json:"name" gorm:"primaryKey"`
 	CreatedAt time.Time `json:"createdAt"`
 	UpdatedAt time.Time `json:"updatedAt"`
-	// GormParameters are only used by Gorm to persist parameters as it cannot persist a
-	// StackParameters. Only use GormParameters within the repository. Otherwise use
-	// Parameters.
-	GormParameters []StackParameter `json:"parameters" gorm:"foreignKey:StackName; references: Name; constraint:OnUpdate:CASCADE,OnDelete:CASCADE"`
+	// GormParameters are only used by Gorm to persist parameters as it cannot persist a StackParameters.
+	// Only use GormParameters within the repository. Otherwise, use Parameters.
+	GormParameters []StackParameter `json:"parameters" gorm:"foreignKey:StackName; references:Name; constraint:OnUpdate:CASCADE,OnDelete:CASCADE"`
 	// Parameters used by the stacks helmfile template.
-	Parameters       StackParameters `json:"-" gorm:"-"`
-	Instances        []Instance      `json:"instances" gorm:"constraint:OnUpdate:CASCADE,OnDelete:CASCADE;"`
+	Parameters       StackParameters `json:"-" gorm:"-:all"`
+	Instances        []Instance      `json:"instances" gorm:"constraint:OnUpdate:CASCADE,OnDelete:CASCADE"`
 	HostnamePattern  string          `json:"hostnamePattern"`
 	HostnameVariable string          `json:"hostnameVariable"`
 	// Providers provide parameters to other stacks.
 	Providers Providers `json:"-" gorm:"-"`
 	// Requires these stacks to deploy an instance of this stack.
-	Requires []Stack `json:"-" gorm:"-"`
+	Requires []Stack `json:"requires" gorm:"constraint:OnUpdate:CASCADE,OnDelete:CASCADE"`
```

Comment:
I would not yet expose this in our JSON and also not yet persist it until its needed. So we can refactor freely.